src/vue/fields/DateField.vue

```diff
@@ -93,139 +119,226 @@ export default {
   },
 
   watch: {
-    'value': function () {
-      this.flatpickrDate = this.value
+    /**
+     * Watch for changes from parent component and update DOM
+     *
+     * @param {string|date} newValue
+     */
+    value (newValue) {
+      // Prevent updates if v-model value is same as input's current value
+      if (newValue === this.flatpickrDate) return
+      // Sets the current selected date after value changed
+      if (this.fp) this.fp.setDate(newValue, true)
+    },
+    /**
+     * Watch for any config changes and redraw date-picker
+     *
+     * @param {object} newConfig
+     */
+    config: {
+      deep: true,
+      handler (newConfig) {
+        let safeConfig = Object.assign({}, newConfig)
+        // Workaround: Don't pass hooks to configs again otherwise
+        // previously registered hooks will stop working
+        // Notice: we are looping through all events
+        // This also means that new callbacks cannot be passed once a component
+        // has been initialized
+        ALLOWED_FLATPICKR_EVENTS.forEach((hook) => {
+          delete safeConfig[hook]
+        })
```

Comment:
if we extract this part to method like "removeFlatpickrHooks" we can get rid of half of the comment above